PreetiDantkale/java_training
# Sample maven project

Notes:

//maven
Project management Tool
It helps to:
1. Build
2. Documentation
3. Reporting
4. SCM
5. Releases & Distribution

Maven Repo is
1. Directory of package jar with pom.xml file
2. Searches for dependencies in repos
maven repos
1. local - locally
2. central - located on the web, created by apache maven
3. remote - located on the web. most of the missing libraries in the central repo can be found here
If the dependency is not found anywhere then throws an error.

1. Pom - project obj model

Elements
1. project
3. modelVersion
4. groupID
5. artifactId
6. version

install maven in ubuntu => "https://linuxize.com/post/how-to-install-apache-maven-on-ubuntu-20-04/"
archetype is a template, which creates a project structure
maven works on the plugin.

cmd: mvn archetype:generate
src-> main
    -> test

maven build lifecycle
1. validate - mvn validate
2. compile - mvn compile
3. test - mvn test
4. package -mvn package -> take the compiled code & package it in a distributable format
5. verify - mvn verify -> runs check to ensure quality criteria are met
7. install - install package in a local repo, for use in other repos as well
8. deploy - done in build env, copies final package to remote repo for sharing with other developers

commands
mvn archetype:generate
cd java -cp maven-demo-1.0-SNAPSHOT.jar com.pd.javamaven.App

Issues before maven
1. set of jars
2. right project structure
3. building & deploying project
